pkg/controller/deactivation/deactivation_controller.go

```diff
@@ -58,6 +73,7 @@ var _ reconcile.Reconciler = &ReconcileDeactivation{}
 
 // ReconcileDeactivation reconciles a Deactivation object
 type ReconcileDeactivation struct {
+	*usersignup.StatusUpdater
```

Comment:
TBH, I have a bad feeling that this will break the "primary/secondary resource" approach
The status of the `UserSignup` should be modified only by the `UserSignup_controller` (`UserSignup` is a primary resource there). The spec should be modified only by the clients - which is also the `Deactivation_controller`. 
However, with this change the deactivation controller would modify both - the status and the spec.

Reply:
What I was trying to say was that the `UserSignup_controller` should take care of sending the notification and updating the status. By that it will notify the clients (`deactivation_controller`) that the notification was already sent and based on that information the `deactivation_controller` can start counting the last 3 days before setting the `UserSignup.spec.deactivated` to `true`

Reply:
Another approach would be prolonging the lifetime of the `Notification` CR to 4 days. Then the `deactivation_controller` could check if it exists or not. In the future, we could add a parameter to the `Notification` CRD that would set the lifetime of the CRs. As a downside I see that we will lose the information from the `UserSignup.status`, which could be a complication in a case we would need to troubleshoot something :-/.
Or, we could create a new CRD that would keep the track of the UserSignup deactivation.

Reply:
So we discussed it during our sync meeting, and we agreed that prolonging `Notification` CR is a bad idea - so let's not consider this option anymore.
We discussed two more options:
1. instead of storing the information in `UserSignup.status` we could store it as an annotation of the `UserSignup` CR. In that case the deactivation controller would manage the whole deactivation flow and would also update the `UserSignup` CR.
2. second options was splitting the notification status info into `UserSignup.status` & `MUR.status`. `UserSignup.status` would still keep the information that the "deactivated notification" was sent and `MUR.status`would keep the information about "deactivating notification". That would mean that MUR controller would need to monitor the 27 days before sending the "deactivating notification" and also would be responsible for creating the notification CR.

Personally, I would vote for the first options - storing the information as a `UserSignup` annotation, but it depends on the amount of work & refactoring. Looking at the code, I believe that it will require even less work that the second option.

Reply:
I'm also for the first option (annotations) if it doesn't require significantly more work than the second one.

Reply:
Out of those two options I prefer the first one also, however I've been contemplating whether there might be another option besides these two, however the ideas I'm working through would require a redesign.  This might end up being dependent on the notification subsystem redesign too, @alexeykazakov will probably discuss this with you tomorrow during our weekly call.

Reply:
Guys, I'm going to implement this using the new states property, however this doesn't mean we require a full refactor of all the existing usage of `Approved`, `Deactivated`, `VerificationRequired` properties right now.  We can use a hybrid approach here, making use of the `states` property only for the new `Deactivating` state, then followup after this with separate refactorings for the other properties.  I'll create a new story in Jira to track this.